sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosRowConverter.scala

```diff
@@ -100,39 +100,75 @@ private class CosmosRowConverter(
 
     def fromRowToObjectNode(row: Row): ObjectNode = {
 
-        if (row.schema.contains(StructField(CosmosTableSchemaInferrer.RawJsonBodyAttributeName, StringType))){
-            // Special case when the reader read the rawJson
-            val rawJson = row.getAs[String](CosmosTableSchemaInferrer.RawJsonBodyAttributeName)
-            objectMapper.readTree(rawJson).asInstanceOf[ObjectNode]
-        }
-        else {
-            val objectNode: ObjectNode = objectMapper.createObjectNode()
-            row.schema.fields.zipWithIndex.foreach({
-              case (field, i) =>
-                field.dataType match {
-                  case _: NullType => putNullConditionally(objectNode, field.name)
-                  case _ if row.isNullAt(i) => putNullConditionally(objectNode, field.name)
-                  case _ =>
-                    val nodeOpt = convertSparkDataTypeToJsonNode(field.dataType, row.get(i))
-                    if (nodeOpt.isDefined) {
-                      objectNode.set(field.name, nodeOpt.get)
-                    }
+      val rawBodyFieldName = if (row.schema.names.contains(CosmosTableSchemaInferrer.RawJsonBodyAttributeName) &&
+        row.schema.apply(CosmosTableSchemaInferrer.RawJsonBodyAttributeName).dataType.isInstanceOf[StringType]) {
+        Some(CosmosTableSchemaInferrer.RawJsonBodyAttributeName)
+      } else if (row.schema.names.contains(CosmosTableSchemaInferrer.OriginRawJsonBodyAttributeName) &&
+        row.schema.apply(CosmosTableSchemaInferrer.OriginRawJsonBodyAttributeName).dataType.isInstanceOf[StringType]) {
+        Some(CosmosTableSchemaInferrer.OriginRawJsonBodyAttributeName)
+      } else {
+        None
+      }
+
+      if (rawBodyFieldName.isDefined){
+
+        val rawBodyFieldIndex = row.schema.fieldIndex(rawBodyFieldName.get)
+        // Special case when the reader read the rawJson
+        val rawJson = row.getAs[String](rawBodyFieldName.get)
+        jsonToObjectNode(rawJson, rawBodyFieldName.get == CosmosTableSchemaInferrer.OriginRawJsonBodyAttributeName)
+      } else {
+        val objectNode: ObjectNode = objectMapper.createObjectNode()
+        row.schema.fields.zipWithIndex.foreach({
+          case (field, i) =>
+            field.dataType match {
+              case _: NullType => putNullConditionally(objectNode, field.name)
+              case _ if row.isNullAt(i) => putNullConditionally(objectNode, field.name)
+              case _ =>
+                val nodeOpt = convertSparkDataTypeToJsonNode(field.dataType, row.get(i))
+                if (nodeOpt.isDefined) {
+                  objectNode.set(field.name, nodeOpt.get)
                 }
-            })
+            }
+        })
 
-            objectNode
-        }
+        objectNode
+      }
+    }
+
+    private def jsonToObjectNode(json: String, keepOriginProperties: Boolean): ObjectNode = {
```

Comment:
nit: the method name is a bit obscure (it rings similar to the other conversion methods). Maybe something that mentions specifically that is for handling raw Json?

Reply:
Done (next iteration)